test/integration/version_upgrade_test.go

```diff
@@ -35,6 +35,113 @@ import (
 	pkgutil "k8s.io/minikube/pkg/util"
 )
 
+// TestDifferentVersionUpgrade test installation of different version of Kubernetes.
+// Start with installing from v1.13.0 then to v1.15.0 and then try to downgrade it
+// to v1.13.0
+// Following are the test cases that the function is doing
+// 1. Start with specific version
+//    ./out/minikube start -p vupgrade --kubernetes-version=v1.13.0
+// 2. Upgrade with newer version
+//   ./out/minikube start -p vupgrade --kubernetes-version=v1.13.0
+// 3. Downgrade with older version (this will spit out error and test the error message)
+//   ./out/minikube start -p vupgrade --kubernetes-version=v1.13.0
+func TestDifferentVersionUpgrade(t *testing.T) {
+	MaybeParallel(t)
+	WaitForStartSlot(t)
+	profile := UniqueProfileName("testkubernetestversions")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 55*time.Minute)
+	defer CleanupWithLogs(t, profile, cancel)
+
+	// create a temporary file
+	tf, err := ioutil.TempFile("", "minikube-release.*.exe")
+	if err != nil {
+		t.Fatalf("tempfile: %v", err)
+	}
+	defer os.Remove(tf.Name())
+	tf.Close()
+
+	// download minikube
+	url := pkgutil.GetBinaryDownloadURL("latest", runtime.GOOS)
+
+	// chmod the binary (when Linux)
+	if runtime.GOOS != "windows" {
+		if err := os.Chmod(tf.Name(), 0700); err != nil {
+			t.Errorf("chmod: %v", err)
+		}
+	}
+
+	// download the file
+	if err := retry.Expo(func() error { return getter.GetFile(tf.Name(), url) }, 3*time.Second, 3*time.Minute); err != nil {
+		t.Fatalf("get failed: %v", err)
+	}
+
+	t.Run("parallel", func(t *testing.T) {
+		tests := []struct {
+			name                 string
+			message              string
+			kubernetesversion    string
+			expectstartupfailure bool
+		}{
+			{"fresh install kubernetes v1.13.0", "v1.13.0", "v1.13.0", false},
+			{"upgrade kubernetes to v 1.14.0", "v1.14.0", "v1.14.0", false},
+			{"downgrade kubernetes to v 1.12.0", "Non-destructive downgrades are not supported", "v1.12.0", true},
+		}
```

Comment:
Please don't hardcode particular Kubernetes versions in tests when possible - they get to be a pain to maintain when we drop old versions.